src/reason_pprint_ast.ml

```diff
@@ -691,6 +691,16 @@ let default = new Pprintast.printer ()
 type funcReturnStyle =
   | ReturnValOnSameLine
 
+let rec detectJSXComponent e attributes =
+  match (e, attributes) with
+    | (Pexp_ident loc, ({txt = "JSX"; loc = _}, PStr []) :: tail) ->
```

Comment:
You may also want to make sure the txt part of the "loc" has a final component of "createElement". Otherwise `Foo.bar [] [@JSX]` will also be formatted as `<Foo />;`